### PR TITLE
Fix reviewers getting removed on breadcrumb update and browser open command bitbucket

### PR DESCRIPTION
**Pull request by mw00120**
Hello! Thanks for maitaining git-town, it's an awesome tool, I use it all the time 👍 

Some context on my setup: I use windows and bitbucket cloud

I had two painpoints which were causing issues

1. When I would do git town propose on a stacked PR (i.e. not against main), the "browser open" would fail out saying `dest` is not a recognized program: (It tries to open `<url>&dest=target-branch`). This issue has existed forever, but I figured I would try to fix it since my second problem was much more serious.
2. I recently upgraded to 24.0 (from something ~21, there are some great new features!) and enabled the `breadcrumb` feature, which works fine however when I would do a sync on a stack of PRs, all my reviewers would get removed :( 

I'm not really a go programmer, so I tried some debugging and fixing with AI and was able to get both pain-points resolve with what seem like reasonable changes... It was having a lot of trouble figuring out why the escape of the `&` didn't work and decided the easiest fix was to use:

```
rundll32 url.dll,FileProtocolHandler url
```

Changing the program used here seems odd to me, but apparently this is equivalent to asking the OS for the "configured browser" and opening using that, and ships in all windows installs... If you think this is the wrong approach I can try to make the actual escaping work instead..

As for wiping the reviewer, it ended up being kind of similar to https://github.com/git-town/git-town/issues/4900 where the body being sent in the PUT to update the PR did not have all the information from the PR loaded into it and the PUT defaults to clearing out the fields that are unspecified.. The solution was to load the particular PR's data and populate the PUT accordingly!

I have a build of this working on my machine, and I haven't seen any other regressions so far.. Let me know if you have any feedback I'm happy to help make it work for other windows and bitbucket cloud users (maybe only me 😆 )

### CODE REVIEW

**Comment by mw00120** [comment]
Actually I just found an edge case I think

When running `git town delete` on the branch I was using to test, I get the following log to finish:

```
Update body for #1683 ... 400 Bad Request
```

and the proposal isn't removed.. I will see if that's related to my changes or already in 24.0

**Comment by kevgo** [comment]
Thanks for fixing these issues! This PR addresses several bugs. Since the changes are fairly complex and I’d like to verify each fix independently before merging, could you please submit each bug fix on a separate branch?

My Windows laptop just died, so I’ll need to wait a few days until I get a replacement before I can properly review and test these changes.

For bug 1, I’m hoping we can fix it by encoding the URL we pass to the browser more carefully, or possibly by escaping the special characters in it. If that's not possible, we can go with your solution.

Bug 2 looks like a legitimate bug in our Bitbucket connector.

**Comment by mw00120** [comment]
Yep, no problem! Will try to do that soon here.. since you don't have the resources to test right now, I will check that out at the start of next week :) 

I agree with your take on the encoding issue.. I will see if I can fix that

I have marked this PR as a draft so I don't forget, and I will close it once I open the other ones

File: vendor/github.com/ktrysmt/go-bitbucket/pullrequests.go [hunk 1]
```diff
@@ -218,35 +218,49 @@ func (p *PullRequests) Statuses(po *PullRequestsOptions) (interface{}, error) {
...
-	body["destination"] = map[string]interface{}{}
-	body["reviewers"] = []map[string]string{}
-	body["title"] = ""
-	body["description"] = ""
-	body["message"] = ""
-	body["close_source_branch"] = false
-
-	if n := len(po.Reviewers); n > 0 {
+	if n := len(po.ReviewerAccountIDs); n > 0 {
+		body["reviewers"] = make([]map[string]string, n)
```
Comment (mw00120) [inline vendor/github.com/ktrysmt/go-bitbucket/pullrequests.go:222]: oh I didn`t even see that this is a vendored package.. do you want me to submit a patch there or are you ok with me patching directly here @kevgo

**Comment by mw00120** [comment]
closed in favor of two targetted PRs

File: vendor/github.com/ktrysmt/go-bitbucket/pullrequests.go (hunk 1 above)
Comment (kevgo) [reply vendor/github.com/ktrysmt/go-bitbucket/pullrequests.go:222]: @mw00120 whoops I missed that too... Let's do the right thing and submit the patch at the upstream repo. I update vendored dependencies regularly, so any patches to the vendored libraries will get lost. Thanks!
